one-ui/src/main/java/com/marklogic/hub/curation/controllers/FlowController.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2012-2019 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.hub.curation.controllers;
+
+import com.marklogic.hub.curation.services.FlowManagerService;
+import com.marklogic.hub.error.DataHubProjectException;
+import com.marklogic.hub.flow.Flow;
+import com.marklogic.hub.oneui.exceptions.DataHubException;
+import com.marklogic.hub.oneui.models.StepModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping("/api/flows")
+@EnableAsync
+public class FlowController {
+
+    @Autowired
+    private FlowManagerService flowManagerService;
+
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity<?> createFlow(@RequestBody String flowJson) {
+        Flow flow;
+        try {
+            flow = flowManagerService.createFlow(flowJson, false);
+            if (flow == null) {
+                throw new DataHubException("Flow request payload is invalid.");
+            }
+        } catch (Exception dpe) {
+            throw new DataHubException(dpe.getMessage());
+        }
+        return new ResponseEntity<>(flow, HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/{flowName}", method = RequestMethod.PUT)
+    @ResponseBody
+    public ResponseEntity<?> updateFlow(@PathVariable String flowName, @RequestBody String flowJson) {
+        Flow flow;
+        try {
+            flow = flowManagerService.createFlow(flowJson, true);
```

Comment:
Same thing here, createFlow should throw a useful exception. There shouldn't be any need for the controller to do anything other than call the service method.